specs/Standards/StorageManagement.md

```diff
@@ -0,0 +1,246 @@
+
+   Response:
+
+       View call: mochi.storage_balance_of({"account_id": "bob"})
+       {
+         total: '4700000000000000000000',
+         available: '2350000000000000000000'
+       }
+
+2. Alice calls `mochi::storage_withdraw({"amount": "2350000000000000000000"})` for her own account. NEAR CLI command:
+
+       near call mochi storage_withdraw \
+         '{"amount": "2350000000000000000000"}' \
+         --accountId alice
+
+3. Alice realizes that `storage_withdraw` does not allow specifying the account to withdraw from. She has withdrawn all she can from her own account. When she re-checks `mochi::storage_balance_of({"account_id": "alice"})`, it indicates she has zero available balance. This is because storage withdrawal is only for the predecessor account that has signed the transaction. Alice cannot withdraw for Bob.
+
+4. Bob issues the same transaction as Alice did in step 2.
+
+       near call mochi storage_withdraw \
+         '{"amount": "2350000000000000000000"}' \
+         --accountId bob
+
+## Reference-level explanation
+
+**NOTES**:
+
+- All amounts, balances and allowance are limited by `U128` (max value 2<sup>128</sup> - 1).
+- This storage standard uses JSON for serialization of arguments and results.
+- Amounts in arguments and results are serialized as Base-10 strings, e.g. `"100"`. This is done to avoid JSON limitation of max integer value of 2<sup>53</sup>.
+- To prevent the deployed contract from being modified or deleted, it should not have any access keys on its account.
+
+**Interface**:
+
+```ts
+// The below defines the structure that will be returned for the methods:
+// * `storage_deposit`
+// * `storage_withdraw`
+// * `storage_balance_of`
+// The `total` and `available` values are string representations of unsigned
+// 128-bit integers.
+type AccountStorageBalance = {
+   total: string;
+   available: string;
+}
+
+/************************************/
+/* CHANGE METHODS on fungible token */
+/************************************/
+// Payable method that receives an attached deposit of Ⓝ for a given account.
+// Requirements:
+// * `account_id`, if provided, must be a valid account name.
+// * The account sending the transaction must have enough Ⓝ to cover the
+//   deposit.
+// If `account_id` is omitted, the sender's account will receive the attached
+// deposit for storage.
+// Returns the AccountStorageBalance structure
+function storage_deposit(
+    account_id: string|null
+): AccountStorageBalance {}
+
+// Withdraws a specified amount of Ⓝ for a given account that was previously
+// reserved for storage.
+// Requirements:
+// * `amount` must be less than or equal to the available storage balance for
+//   the sending account. Is sent as a string but represents an unsigned
+//   128-bit integer.
+// * The calling account must already be registered with the contract
+// Returns the AccountStorageBalance structure
+function storage_withdraw(
+    amount: string
```

Comment:
If `amount == null` then the all available storage balance should be withdrawn

```suggestion
    amount: string|null
```

Reply:
this is addressed now